src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,248 @@
+    class RecordCollection
+      include Enumerable
+
+      def initialize(record_class, augeas_collection)
+        @record_class = record_class
+        @augeas_collection = augeas_collection
+      end
+
+      def each
+        @augeas_collection.each do |augeas_value|
+          yield @record_class.new_from_augeas(augeas_value)
+        end
+      end
+
+      def add(record)
+        @augeas_collection.add(record.to_augeas)
+      end
+
+      def delete(record)
+        @augeas_collection.delete(record.to_augeas)
+      end
+
+      def replace(old_record, new_record)
+        return if !include? old_record
+        matcher = CFA::Matcher.new { |_k, v| v == old_record.to_augeas }
+        placer = CFA::ReplacePlacer.new(matcher)
+        @augeas_collection.add(new_record.to_augeas, placer)
+      end
+
+      def empty?
+        @augeas_collection.empty?
+      end
+
+      def ==(other)
+        other.to_a == to_a
+      end
+    end
+
+    # class to represent a general entry of the file.
+    class Record
+      def initialize(value: nil, tree_data: [])
+        @value = value
+        @tree_data = tree_data
+      end
+
+      def self.new_from_augeas(augeas_value)
+        value, tree_data = parse_augeas(augeas_value)
+        record = new
+        record.send(:add_value, value)
+        record.send(:add_tree_data, tree_data)
+        record
+      end
+
+      attr_reader :value
+
+      def to_augeas
+        create_augeas
+      end
+
+      def comment
+        comment = @tree_data.select { |d| d[:key] == "#comment" }.first
+        comment[:value] if !comment.nil?
+      end
+
+      def ==(other)
+        other.class == self.class &&
+          other.value == value &&
+          other.tree_data == tree_data
+      end
+
+      alias_method :eq?, :==
+
+      protected
+
+      attr_reader :tree_data
+
+      def self.parse_augeas(augeas_value)
+        case augeas_value
+        when CFA::AugeasTreeValue
+          value = augeas_value.value
+          tree_data = augeas_value.tree.data
+        else
+          value = augeas_value
+          tree_data = []
+        end
+        [value, tree_data]
+      end
+
+      private_class_method :parse_augeas
+
+      private
+
+      def add_value(value)
+        @value = value
+      end
+
+      def add_tree_data(tree_data)
+        @tree_data = tree_data
+      end
+
+      def add_comment(comment)
+        @tree_data << { key: "#comment", value: comment } if !comment.nil?
+      end
+
+      def create_augeas
+        return value if @tree_data.empty?
+        tree = CFA::AugeasTree.new
+        @tree_data.each { |d| tree.add(d[:key], d[:value]) }
+        CFA::AugeasTreeValue.new(tree, value)
+      end
+    end
+
+    # class to represent a key-value ntp entry.
+    #   For example:
+    #     logfile /var/log/ntp
+    class AttributeRecord < Record
+      def initialize(value: nil, comment: nil)
+        super()
+        add_value(value)
+        add_comment(comment)
```

Comment:
add_comment is private, not protected. Also it looks a bit strange to have child that have so different arguments in constructor